packages/db/src/connection.ts

```diff
@@ -71,30 +105,7 @@ const buildKnexConfiguration = ({
 * @returns {Promise<Knex>} Returns a configured knex instance
 */
 export const knex = async (env: NodeJS.ProcessEnv): Promise<Knex> => {
-  let connectionConfig: Knex.PgConnectionConfig;
-
-  if (env.databaseCredentialSecretArn) {
-    const secretsManager = new AWS.SecretsManager();
-    connectionConfig = await getSecretConnectionConfig(
-      env.databaseCredentialSecretArn,
-      secretsManager
-    );
-  } else {
-    connectionConfig = getConnectionConfigEnv(env);
-  }
-
-  let timeout;
-  if (env.knexAcquireConnectionTimeout) {
-    timeout = Number(env.knexAcquireConnectionTimeout);
-  }
-
-  const knexConfig = buildKnexConfiguration({
-    connectionConfig,
-    asyncStackTraces: env.KNEX_ASYNC_STACK_TRACES === 'true',
-    debug: env.KNEX_DEBUG === 'true',
-    migrationDir: env.migrationDir,
-    timeout,
-  });
-
+  const connectionConfig = await getConnectionConfig(env);
```

Comment:
@Jkovarik How do you feel about this refactor? It allows me to easily use the connection config to get the hostname here:

https://github.com/nasa/cumulus/pull/1828/files?file-filters%5B%5D=.ts#diff-88fa195f355a7e766958b02120bd5a2aR66

It also doesn't break any existing code

Reply:
It's probably 'ok' - it's doubling the AWS SM cost of establishing a connection ($0.05 per 10,000 API calls), but only for provisioning that even cares about that info , that shouldn't cause heartburn. 

We should be careful we don't adopt this pattern on scaleable processes though, or if we do, memoizing the call or some other caching solution.

Reply:
How is it doubling the cost? The same amount of calls to retrieve the secret are made. I just broke down this function into two smaller functions so that in the provision database lambda I can inspect the connection config and then turn around and use it to build a Knex connection

Reply:
Right you are - when I read this yesterday I somehow thought `getKnexConfig` was re-making a call to SM.    Ignore the above, that's entirely 💯